src/backend/optimizer/path/allpaths.c

```diff
@@ -2125,15 +2125,25 @@ set_cte_pathlist(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
 			subplan = subquery_planner(cteroot->glob, subquery, cteroot, cte->cterecursive,
 									   tuple_fraction, &subroot, config);
 
-			cteplaninfo->shared_plan = prepare_plan_for_sharing(cteroot, subplan);
+			cteplaninfo->subplan = (subplan->flow->locustype != CdbLocusType_SegmentGeneral) ?
```

Comment:
use `if` instead of `?` because later may need to add one another condition and with the operator `?` it won't look good anymore.

Reply:
It can be easily rewritten when the need for another condition comes

Reply:
I still think `if` is more readable.

Reply:
Put `if` instead. Because your conditions will be extended for General locus in additional patch, and it would really be more readable.